json_exerciseAnswers.ipynb

```diff
@@ -0,0 +1,619 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "People's Republic of China         19\n",
+       "Republic of Indonesia              19\n",
+       "Socialist Republic of Vietnam      17\n",
+       "Republic of India                  16\n",
+       "Republic of Yemen                  13\n",
+       "Kingdom of Morocco                 12\n",
+       "People's Republic of Bangladesh    12\n",
+       "Nepal                              12\n",
+       "Republic of Mozambique             11\n",
+       "Africa                             11\n",
+       "Name: countryname, dtype: int64"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#Here is the answer to question 1\n",
+    "#read in data as data frame\n",
+    "data = pd.read_json('data/world_bank_projects.json')\n",
+    "\n",
+    "#pull out top 10 countries with most projects\n",
+    "answer1 = data.countryname.value_counts().head(10)\n",
+    "\n",
+    "#print answer\n",
+    "print(answer1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Environment and natural resources management    223\n",
+      "Rural development                               202\n",
+      "Human development                               197\n",
+      "Public sector governance                        184\n",
+      "Social protection and risk management           158\n",
+      "Financial and private sector development        130\n",
+      "Social dev/gender/inclusion                     119\n",
+      "Trade and integration                            72\n",
+      "Urban development                                47\n",
+      "Economic management                              33\n",
+      "Name: name, dtype: int64\n"
+     ]
+    }
+   ],
+   "source": [
+    "#Question 2:\n",
+    "#read in the data as a json\n",
+    "data = json.load((open('data/world_bank_projects.json')))\n",
+    "# use normalization to create tables from nested element\n",
+    "df = json_normalize(data, 'mjtheme_namecode')\n",
+    "\n",
+    "# get rid of any rows that don't have a project theme\n",
+    "result = df[df.name != \"\"]\n",
+    "\n",
+    "# pull out top 10 project themes\n",
+    "results = result.name.value_counts().head(10)\n",
+    "\n",
+    "#print results\n",
+    "print(results)\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#Question 3:\n",
+    "#read in the data as a json\n",
+    "data = json.load((open('data/world_bank_projects.json')))\n",
```

Comment:
A general point:
[DRY](https://www.wikiwand.com/en/Don't_repeat_yourself) is a very useful concept in software development. Every time you write something more than once you should think about it. It's easy to overdo it too so don't worry about it for this exercise but just be aware of its existence.